src/CFG-based-system.typ

```diff
@@ -1 +1,178 @@
-= CFG Based System
\ No newline at end of file
+#import "./proof-tree.typ": *
+#import "./vars.typ": *
+= CFG Based System
+
+Basic rules for Uniqueness and Borrowing won't change here,
+but adapted to fit the CFG representation of Kotlin.
+
+The key gap between the CFG version and the type rules is at the function call.
+Type checking at function calls need adjustment so that it can be done in control flow order.
+The support for recursive function all is alse missing is previous typing rules.
```

Comment:
nit: recursive function calls are when `f` calls `f`.  What you mean here are nested function calls, i.e. cases like  `f(g(x))`.